wled00/wled_server.cpp

```diff
@@ -15,6 +15,22 @@ bool isIp(String str) {
   return true;
 }
 
+void handleUpload(AsyncWebServerRequest *request, const String& filename, size_t index, uint8_t *data, size_t len, bool final){
+  if(!index){
+    request->_tempFile = WLED_FS.open(filename, "w");
+    DEBUG_PRINT("Uploading ");
+    DEBUG_PRINTLN(filename);
+    if (filename == "/presets.json") presetsModifiedTime = toki.second();
+  }
+  if (len) {
+    request->_tempFile.write(data,len);
+  }
+  if(final){
+    request->_tempFile.close();
```

Comment:
It appears possible to destroy the configuration, if either no data is sent, or the transfer is interrupted before completed (e.g., WiFi hiccup causes disconnect at inopportune time.)

Does the WLED_FS interface allow renaming an existing file?

If so, I would like to recommend a slightly more complex, but fail-safe, method:

1. For each allowed file, pre-define an intermediate file name that is used for the upload.  (e.g., for `/presets.json`, name could be `/presets.json.uploading`)
2. Upload only goes to these intermediate filenames
3. Optionally validate the contents (function pointer for each allowed filename)
4. Delete the target file from the file system (e.g., `/presets.json`)
5. Rename the uploaded file (e.g. `/presets.json/uploading` --> `/presets.json`)

In addition, at boot, perform the following steps for each target filename in the allowed upload list:
1. If the intermediate file exists:
    A. If the target filename exists, delete the intermediate file (failed or interrupted upload)
    B. Else (target filename does not exist), rename the intermediate file to the target filename

In this way, the upload process will have many fewer failure points.

Reply:
It could be possible as it is possible to remove files currently using /edit page. Nothing new here.
Assuming you use Backup & Restore functions, normal behavior would be to first back-up configuration (and later restore it if needed).
If there is an adversary trying to sabotage your WLED (assuming you let him/her into your network) he/she can easily do it even without backup/restore functions.
If you have WLED on a publicly accessible network, ... well, he/she can wreck havoc just changing effects or overwriting/deleting presets even with password lock.

Nevertheless I agree on your comments and suggestions.